src/dataflow/src/render/delta_join.rs

```diff
@@ -73,16 +75,54 @@ where
                             offset += arities[input];
                         }
 
+                        // filter arrangements using semijoins if possible
+                        let mut equivalences = equivalences.clone();
+                        for equivalence in equivalences.iter_mut() {
+                            equivalence.sort();
+                            equivalence.dedup();
+                        }
```

Comment:
I'm not clear on how the comment relates to the code. This looks to be normalizing `equivalances` more than anything else?